chat/viewsets.py

```diff
@@ -61,6 +61,27 @@ def star_message(self, request, pk=None, msg_id=None):
         message.save()
         return Response({"message": f"Message starred: {message.starred}"})
 
+    @action(detail=False, methods=['POST'])
+    def sql_query(self, request):
```

Comment:
Good start, as I said above, this should be a 2 part process.

Step 1. Load in a db and create the schema for it.
Step 2. Use that schema to build the query.

Big thing here is gonna be getting a nice schema that can express the db arch well.